From 55d312e6a3d9c5bdf5ae20bb1e2bad98edc38f03 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 18 Mar 2026 04:28:35 +0000
Subject: [PATCH 10/15] Remove references to excluded apps from framework

Remove Auditor references:
- Remove app.attestation.auditor from GmcPackageManager HIDDEN_PACKAGES
- Remove bundled Auditor app certificate digest from
  read_additional_security_state_known_signers
- Remove app.attestation.auditor from notification permission grants
Auditor relies on hardware attestation and will never pass on a GSI.

Remove SELinux flags kernel support notification:
The GSI does not ship a GrapheneOS-patched kernel so the custom
SELinux flags proc attribute will never be present. Remove the
persistent notification since it is expected and not actionable.

Remove MusicFX references:
MusicFX is excluded from the build. Remove stale hidden API whitelist
and privileged app permissions entries.
---
 .../internal/gmscompat/sysservice/GmcPackageManager.java  | 1 -
 core/res/res/values/config_ext.xml                        | 2 --
 data/etc/hiddenapi-package-whitelist.xml                  | 2 +-
 data/etc/privapp-permissions-platform.xml                 | 4 ----
 .../core/java/com/android/server/ext/SystemServerExt.java | 8 --------
 .../pm/permission/DefaultPermissionGrantPolicy.java       | 1 -
 6 files changed, 1 insertion(+), 17 deletions(-)

diff --git a/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java b/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
index 3fe16a1a2..6476ed372 100644
--- a/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
+++ b/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
@@ -234,7 +234,6 @@ public class GmcPackageManager extends ApplicationPackageManager {
     }
 
     private static final ArraySet<String> HIDDEN_PACKAGES = new ArraySet<>(new String[] {
-            "app.attestation.auditor",
     });
 
     private static void throwIfHidden(String pkgName) throws NameNotFoundException {
diff --git a/core/res/res/values/config_ext.xml b/core/res/res/values/config_ext.xml
index 9f66d22bc..8264085d1 100644
--- a/core/res/res/values/config_ext.xml
+++ b/core/res/res/values/config_ext.xml
@@ -46,8 +46,6 @@
          The digest should be computed over the DER encoding of the trusted certificate
          using the SHA-256 digest algorithm. -->
     <string-array name="read_additional_security_state_known_signers">
-        <!-- bundled Auditor app certdigest -->
-        <item>990e04f0864b19f14f84e0e432f7a393f297ab105a22c1e1b10b442a4a62c42c</item>
     </string-array>
     <java-symbol type="array" name="read_additional_security_state_known_signers" />
 
diff --git a/data/etc/hiddenapi-package-whitelist.xml b/data/etc/hiddenapi-package-whitelist.xml
index 98f582401..1912d314f 100644
--- a/data/etc/hiddenapi-package-whitelist.xml
+++ b/data/etc/hiddenapi-package-whitelist.xml
@@ -41,7 +41,7 @@ Do NOT include any apps that are updatable via Play Store!
   <hidden-api-whitelisted-app package="com.android.gallery" />
   <hidden-api-whitelisted-app package="com.android.launcher3" />
   <hidden-api-whitelisted-app package="com.android.mtp" />
-  <hidden-api-whitelisted-app package="com.android.musicfx" />
+
   <hidden-api-whitelisted-app package="com.android.permissioncontroller" />
   <hidden-api-whitelisted-app package="com.android.printservice.recommendation" />
   <hidden-api-whitelisted-app package="com.android.printspooler" />
diff --git a/data/etc/privapp-permissions-platform.xml b/data/etc/privapp-permissions-platform.xml
index f9d078cb6..613c72c7d 100644
--- a/data/etc/privapp-permissions-platform.xml
+++ b/data/etc/privapp-permissions-platform.xml
@@ -106,10 +106,6 @@ applications that come with the platform
         <permission name="android.permission.MANAGE_EXTERNAL_STORAGE"/>
     </privapp-permissions>
 
-    <privapp-permissions package="com.android.musicfx">
-        <permission name="android.permission.CHANGE_COMPONENT_ENABLED_STATE"/>
-    </privapp-permissions>
-
     <privapp-permissions package="com.android.networkrecommendation">
         <permission name="android.permission.SCORE_NETWORKS"/>
         <permission name="android.permission.SUBSTITUTE_NOTIFICATION_APP_NAME"/>
diff --git a/services/core/java/com/android/server/ext/SystemServerExt.java b/services/core/java/com/android/server/ext/SystemServerExt.java
index ace1e6fd1..c80bdcde1 100644
--- a/services/core/java/com/android/server/ext/SystemServerExt.java
+++ b/services/core/java/com/android/server/ext/SystemServerExt.java
@@ -38,13 +38,5 @@ public final class SystemServerExt {
     void initBgThread() {
         WifiAutoOff.maybeInit(this);
         BluetoothAutoOff.maybeInit(this);
-
-        if (Build.IS_DEBUGGABLE) {
-            if (!SELinuxFlags.kernelSupportsSELinuxFlags()) {
-                String title = "Kernel doesn't support SELinux flags";
-                String msg = "App hardening features that use SELinux flags, such as DCL and ptrace restrictions, do not work.";
-                new SystemErrorNotification("missing hardening", title, msg).show(context);
-            }
-        }
     }
 }
diff --git a/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java b/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
index 053a6bd73..4ca754e99 100644
--- a/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
+++ b/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
@@ -961,7 +961,6 @@ final class DefaultPermissionGrantPolicy {
             "com.android.contacts",
             android.util.PackageUtils.getFirstPartyAppSourcePackageName(mContext),
             "app.grapheneos.camera",
-            "app.attestation.auditor",
             "com.stevesoltys.seedvault",
         };
         for (String pkg : notifPackages) {
-- 
2.53.0

